File: lobby/src/main/java/com/natchuz/hub/lobby/MapConfig.java
package com.natchuz.hub.lobby;

import org.spongepowered.api.world.Location;
import org.spongepowered.api.world.World;

/**
 * Lobby map configuration
 */
public class MapConfig {

    private Location<World> rewards;
    private Location<World> kitpvp;
    private Location<World> spawn;
    private int height;

    /**
     * Get location of rewards npc
     */
    public Location<World> getRewards() {
        return rewards;
    }

    /**
     * Get location of kitpvp npc
     */
    public Location<World> getKitpvp() {
        return kitpvp;
    }

    /**
     * Get spawn location
     */
    public Location<World> getSpawn() {
        return spawn;
    }

    public int getHeight() {
        return height;
    }

    @Override
    public String toString() {
        return "MapConfig{" + "rewards=" + rewards + ", kitpvp=" + kitpvp + '}';
    }
}

File: lobby/src/main/kotlin/com/natchuz/hub/lobby/MapConfig.kt
package com.natchuz.hub.lobby

import org.spongepowered.api.world.Location
import org.spongepowered.api.world.World

/**
 * Lobby map configuration
 */
data class MapConfig(
        /**
         * Get location of rewards npc
         */
        val rewards: Location<World>,
        /**
         * Get location of kitpvp npc
         */
        val kitpvp: Location<World>,
        /**
         * Get spawn location
         */
        val spawn: Location<World>,
        /**
         * Height on which player will teleport on spawn if fall from map
         */
        val height: Int
)

